.github/workflows/release.yml

```diff
@@ -125,11 +128,36 @@ jobs:
           else
             echo "GH_RELEASE_NAME=${{env.TAG_NAME}}" >> $GITHUB_ENV
           fi
-      
-      # PUSH TAGS IF IT IS A RELEASE
-      - name: Push tag if release
+
+      # CREATE PR FOR VERSION AND TAG CHANGES
+      - name: Create Pull Request
+        id: create_pr
+        if: inputs.dev == false
+        uses: peter-evans/create-pull-request@v6.0.5
+        with:
+          token: ${{ env.GITHUB_TOKEN }}
+          branch: ${{ env.BRANCH_NAME }}
+          title: "Update Version - Automated Changes"
+          body: "This is an automated PR created by GitHub Actions"
+          base: main
+          draft: false
+
+      # WAIT FOR PR CREATION
+      - name: Wait for PR to be created
+        id: pr
+        if: inputs.dev == false
+        uses: octokit/request-action@v2.3.1
+        with:
+          route: GET /repos/${{ github.repository }}/pulls?head=${{ github.repository_owner }}:${{ env.BRANCH_NAME }}
+
+      # MERGE PR TO MAIN
+      - name: Merge Pull Request
         if: inputs.dev == false
-        run: git push && git push --tags
+        uses: octokit/request-action@v2.3.1
+        with:
+          route: PUT /repos/${{ github.repository }}/pulls/${{ steps.create_pr.outputs.pull-request-number }}/merge
+          merge_method: squash
```

Comment:
Nice!! you need to delete the branch after if you use the same branch name + for not spamming. I'm not sure if it's automatically happened here or needs additional step

Reply:
the branch is deleted automatically after it is merged.
also if the release flow fails - and we re-run it- the release will use the existing branch and it will pass